CafeteriaRecommendationSystem/Services/SqlService.cs

```diff
@@ -0,0 +1,406 @@
+        }
+
+        public static string GetRecommendedItemsForUser()
+        {
+            string query = @"
+                SELECT r.RecommendationId, r.ItemId, i.Name, i.Price, i.AvailabilityStatus, 
+                s.OverallRating, s.OverallCommentSentiment, 
+                i.DietPreference, i.SpiceLevel, i.FoodPreference, i.SweetTooth
+                FROM Recommendation r
+                JOIN Item i ON r.ItemId = i.ItemId
+                LEFT JOIN Sentiment s ON i.ItemId = s.ItemId
+                JOIN MealType mt ON i.MealTypeId = mt.meal_type_id
+                JOIN UserPreference up ON up.UserId = @UserId
+                WHERE mt.MealType = @MealType
+                ORDER BY 
+                CASE WHEN up.DietPreference = i.DietPreference THEN 1 ELSE 0 END DESC, 
+                CASE WHEN up.SpiceLevel = i.SpiceLevel THEN 1 ELSE 0 END DESC, 
+                CASE WHEN up.FoodPreference = i.FoodPreference THEN 1 ELSE 0 END DESC, 
+                CASE WHEN up.SweetTooth = i.SweetTooth THEN 1 ELSE 0 END DESC";
+            return query;
+        }
+
+        public static void InsertFeedback(MySqlConnection connection, int userId, int itemId, string comment, int rating)
+        {
+            const string query = "INSERT INTO Feedback (UserId, ItemId, Comment, Rating, FeedbackDate) VALUES (@UserId, @ItemId, @Comment, @Rating, NOW())";
+            using (var cmd = new MySqlCommand(query, connection))
+            {
+                cmd.Parameters.AddWithValue("@UserId", userId);
+                cmd.Parameters.AddWithValue("@ItemId", itemId);
+                cmd.Parameters.AddWithValue("@Comment", comment);
+                cmd.Parameters.AddWithValue("@Rating", rating);
+                cmd.ExecuteNonQuery();
+            }
+        }
+
+        public static int UpdateVoteCount(MySqlConnection connection, int itemId)
+        {
+            const string query = "SELECT VoteCount FROM Sentiment WHERE ItemId = @ItemId";
+            using (var cmd = new MySqlCommand(query, connection))
+            {
+                cmd.Parameters.AddWithValue("@ItemId", itemId);
+                var result = cmd.ExecuteScalar();
+                return result != null && result != DBNull.Value ? Convert.ToInt32(result) + 1 : 1;
+            }
+        }
+
+        public static int UpdateVoteCountValue(MySqlConnection connection, int itemId)
+        {
+            const string query = "UPDATE EmployeeVote SET VoteCount = VoteCount + 1 WHERE ItemId = @ItemId";
+            using (var cmd = new MySqlCommand(query, connection))
+            {
+                cmd.Parameters.AddWithValue("@ItemId", itemId);
+                return cmd.ExecuteNonQuery();
+            }
+        }
+
+        public static void InsertNewSentimentData(MySqlConnection connection, int itemId, int rating, string sentimentComment, float sentimentScore, string commentSentiments, int voteCount)
+        {
```

Comment:
Instead of multiple parameters, we can pass an object which consists of all these values.